### PR TITLE
[MI-3651] Add nil checks missing for some metrics in the code

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary
Add nil checks missing for some metrics in the code

### CODE REVIEW

File: server/message_hooks.go [hunk 1]
```diff
@@ -640,8 +642,10 @@ func (p *Plugin) Send(teamID, channelID string, user *model.User, post *model.Po
 	}
 
 	timeElapsed := float64((newMessage.CreateAt.UnixMilli() - post.CreateAt)) / float64(1000)
-	p.metricsService.ObserveMessageSyncDuration(actionSourceMattermost, timeElapsed)
-	p.metricsService.ObserveMessagesCount(actionCreated, actionSourceMattermost, isNotDirectMessage)
+	if p.metricsService != nil {
+		p.metricsService.ObserveMessageSyncDuration(actionSourceMattermost, timeElapsed)
+		p.metricsService.ObserveMessagesCount(actionCreated, actionSourceMattermost, isNotDirectMessage)
```
Comment (raghavaggarwal2308) [inline server/message_hooks.go:647]: I think we changed the name `isNotDirectMessage`. Is this PR not synced with other PRs?

File: server/handlers/attachments.go [hunk 2]
```diff
@@ -147,26 +147,33 @@ func (ah *ActivityHandler) handleAttachments(channelID, userID, text string, msg
...
 		var fileSize int64
 		downloadURL := ""
+		metrics := ah.plugin.GetMetrics()
 		if strings.Contains(a.ContentURL, hostedContentsStr) && strings.HasSuffix(a.ContentURL, "$value") {
 			attachmentData, err = ah.handleDownloadFile(a.ContentURL, client)
 			if err != nil {
 				ah.plugin.GetAPI().LogError("failed to download the file", "filename", a.Name, "error", err.Error())
-				ah.plugin.GetMetrics().ObserveFilesCount(actionCreated, actionSourceMSTeams, isDirect, discardedReasonUnableToGetTeamsData, 1)
+				if metrics != nil {
+					metrics.ObserveFilesCount(actionCreated, actionSourceMSTeams, isDirect, discardedReasonUnableToGetTeamsData, 1)
```
Comment (raghavaggarwal2308) [inline server/handlers/attachments.go:156]: Constant for 1?

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: server/message_hooks.go (hunk 1 above)
Comment (ayusht2810) [reply server/message_hooks.go:647]: yes, we can update it when all the PRs before it gets merged

File: server/handlers/attachments.go (hunk 2 above)
Comment (ayusht2810) [reply server/handlers/attachments.go:156]: same

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/handlers/attachments.go [hunk 3]
```diff
@@ -196,15 +205,21 @@ func (ah *ActivityHandler) handleAttachments(channelID, userID, text string, msg
...
-		ah.plugin.GetMetrics().ObserveFilesCount(actionCreated, actionSourceMSTeams, isDirect, "", 1)
+		if metrics != nil {
+			metrics.ObserveFilesCount(actionCreated, actionSourceMSTeams, isDirect, "", 1)
+		}
 		countFileAttachments++
 		if countFileAttachments == 10 {
 			ah.plugin.GetAPI().LogDebug("discarding the rest of the attachments as Mattermost supports only 10 attachments per post")
-			ah.plugin.GetMetrics().ObserveFilesCount(actionCreated, actionSourceMSTeams, isDirect, discardedReasonFileLimitReached, len(msg.Attachments)-countAttachments-countFileAttachments)
+			if metrics != nil {
+				metrics.ObserveFilesCount(actionCreated, actionSourceMSTeams, isDirect, discardedReasonFileLimitReached, len(msg.Attachments)-countAttachments-countFileAttachments)
```
Comment (Kshitij-Katiyar) [inline server/handlers/attachments.go:221]: can we create this variable and add a comment to denote what is this doing 
len(msg.Attachments)-countAttachments-countFileAttachments

**Review by Kshitij-Katiyar — APPROVED** [review]

**Review by manojmalik20 — APPROVED** [review]

File: server/monitor/monitor.go [hunk 4]
```diff
@@ -11,7 +11,12 @@ import (
 	"github.com/mattermost/mattermost-server/v6/plugin"
 )
 
-const monitoringSystemJobName = "monitoring_system"
+const (
+	monitoringSystemJobName = "monitoring_system"
+	subscriptionConnected   = "connected"
```
Comment (manojmalik20) [inline server/monitor/monitor.go:16]: Please remove these.

**Review by manojmalik20 — APPROVED** [review]

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**